courses/machine_learning/deepdive2/text_classification/labs/classify_text_with_bert.ipynb

```diff
@@ -0,0 +1,1190 @@
+{
```

Comment:
<p>Nit: this could be potentially reworded for further clarity. I read this like the following but could just be me:</p><p> </p><p>"For fine-tuning, let's use the same optimizer that BERT was originally trained with: the "Adaptive Moments" (Adam)." -&gt; ok, so Adam? Thought we imported AdamW above?</p><p> </p><p>This optimizer minimizes the prediction loss and does regularization by weight decay (not using moments), which is also known as <a href="https://arxiv.org/abs/1711.05101" target="_blank">AdamW</a>. -&gt; Ah no ok it was AdamW, not just moments for updates but uses some kind of weight decay so actually AdamW variant.</p><p> </p><p>Since Adam has kinda just been the default across labs, maybe a sentence somewhere mentioning AdamW is a newer optimizer meant to improve generalization</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/1412/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='41'/>

Reply:
<p>Added clarification and also a link to the tensorflow documentation showing how it is implemented.</p>